### PR TITLE
Implement calculator and calculation animation

### PR DESCRIPTION
**Pull request by wixette**
This PR includes:

* Update SeedLang's version. See 49270105aa286d69115b0d1dd14310d852902345
* Update audio, texture and anim resources. See 80ba643b64964efd4b1452b2754eb1541be9403e
* Update packages and build settings. See 4f55e46698b966168de444d3165382f8dc1cffa1
* Add calculator buttons to the main scene. See a9980e3a26b52e3515fa9917121a24e52e94bccb
* Source code update to implement calculator and calculation animation. See f379185673aa983c47760c547b505ac38ac1608e
  * Bind calculator to both the screen and the cutting board
  * Define detailed input behavior of each button (state machine updated)
  * Support animation to demo calculation steps
  * Update the level configs

### CODE REVIEW

File: Assets/Src/Scripts/SeedCalc/CuttingBoard.cs [hunk 1]
```diff
@@ -347,7 +367,12 @@ private class SlideAnimConfig {
       if (texOffsetX > 1.0f) {
         texOffsetX -= 1.0f;
       }
-      GetComponent<Renderer>().material.SetTextureOffset("_MainTex", new Vector2(texOffsetX, 0));
+      GetComponent<Renderer>().material.SetTextureOffset("_MainTex",
+                                                         new UnityEngine.Vector2(texOffsetX, 0));
```
Comment (codingpotato) [inline Assets/Src/Scripts/SeedCalc/CuttingBoard.cs:371]: UnityEngine is not necessary?

File: Assets/Src/Scripts/SeedCalc/CalculatorScreen.cs [hunk 2]
```diff
@@ -118,18 +118,35 @@ public class CalculatorScreen : MonoBehaviour {
...
+      _bottomLine = transform.Find("BottomLine").GetComponent<TextMeshProUGUI>();
       Clear();
     }
 
-    private string GetTokenText(string expression, TextRange range) {
-      return expression.Substring(range.Start.Column, range.End.Column - range.Start.Column + 1);
+    private void PrintTo(TextMeshProUGUI where, string text) {
+      if (string.IsNullOrEmpty(text)) {
+        where.text = _initialText;
+      } else if (!string.IsNullOrEmpty(text)) {
```
Comment (codingpotato) [inline Assets/Src/Scripts/SeedCalc/CalculatorScreen.cs:147]: No need this if check.

File: Assets/Src/Scripts/SeedCalc/Calculator.cs [hunk 3]
```diff
@@ -143,105 +232,229 @@ private class Visualizer : IVisualizer<BinaryEvent>, IVisualizer<EvalEvent> {
...
+        ParsedExpression = new ParsedExpression(expression, syntaxTokens, null, true);
+        if (stepStates.Count > 0) {
+          yield return new WaitForSeconds(_calcAnimInterval);
+        }
+      }
+      ParsedExpression = new ParsedExpression(expression, syntaxTokens, null, false);
+      // Updates the input buffer since the expression could change during the calculation steps.
+      _cachedExpression.Clear();
+      _cachedExpression.Append(expression);
+      // SHows the final result.
```
Comment (codingpotato) [inline Assets/Src/Scripts/SeedCalc/Calculator.cs:446]:
```suggestion
      // Shows the final result.
```

File: Assets/Src/Scripts/SeedCalc/Calculator.cs [hunk 4]
```diff
@@ -143,105 +232,229 @@ private class Visualizer : IVisualizer<BinaryEvent>, IVisualizer<EvalEvent> {
...
+        case CalculatorInput.N2:
+        case CalculatorInput.N3:
+        case CalculatorInput.N4:
+        case CalculatorInput.N5:
+        case CalculatorInput.N6:
+        case CalculatorInput.N7:
+        case CalculatorInput.N8:
+        case CalculatorInput.N9:
+          if (!(ParsedExpression is null) && ParsedExpression.ExtractLastTokenText() == "0") {
+            // If there is only one "0" in the current number, replaces it with the next non-zero
```
Comment (codingpotato) [inline Assets/Src/Scripts/SeedCalc/Calculator.cs:283]: The comment is not correct? Shall be backspace when the last text is "0"?

File: Assets/Src/Scripts/SeedCalc/CalculatorScreen.cs (hunk 2 above)
Comment (wixette) [reply Assets/Src/Scripts/SeedCalc/CalculatorScreen.cs:147]: Done.

File: Assets/Src/Scripts/SeedCalc/Calculator.cs (hunk 4 above)
Comment (wixette) [reply Assets/Src/Scripts/SeedCalc/Calculator.cs]: Done. The comments are updated to a clearer sentence.

**Review by codingpotato — APPROVED** [review]
LGTM

**Pull request merged**